pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -57,9 +59,10 @@ public class PersistentMessageExpiryMonitor implements FindEntryCallback {
             expirationCheckInProgressUpdater = AtomicIntegerFieldUpdater
             .newUpdater(PersistentMessageExpiryMonitor.class, "expirationCheckInProgress");
 
-    public PersistentMessageExpiryMonitor(String topicName, String subscriptionName, ManagedCursor cursor,
-                                          PersistentSubscription subscription) {
-        this.topicName = topicName;
+    public PersistentMessageExpiryMonitor(PersistentTopic topic, String subscriptionName, ManagedCursor cursor,
+                                          @Nullable PersistentSubscription subscription) {
```

Comment:
This PR modifies the public API (even it's in the `pulsar-broker` module), we should not cherry-pick it to release branches unless we make it compatible. @poorbarcode  @Technoboy-